packages/temporal-time-utils/EntityWorkflow.ts

```diff
@@ -0,0 +1,74 @@
+import * as wf from "@temporalio/workflow";
+import { SignalDefinition, QueryDefinition } from "@temporalio/common";
+
+const noop = async () => {};
+type ThingToInvoke =
+  | { activity: string; activityOptions: wf.ActivityOptions }
+  | { workflow: string; workflowOptions: wf.ChildWorkflowOptions };
+export class Entity<Input = any, Update = any> {
+  MAX_ITERATIONS: number;
+  setup: (input: Input) => Promise<void>;
+  cleanup: () => Promise<void>;
+  thingToInvoke: ThingToInvoke;
+  Signal: SignalDefinition<[Update]>;
```

Comment:
Prefer `signal` and `query`.
Also `query` is not well defined here, what would it return? Should there be a generic `State` type too where `query` is actually `getStateQuery`?